### ISSUE TITLE
Groups Graphing

### ISSUE THREAD

**Issue by LeoTindall** [OpenEnergyDashboard/OED#103]
Parallel to #57. Allow graphing between groups (each group is represented in aggregate) or within groups (each meter in a single group is represented individually). 

Design questions:
 - [x] How should the UI look?
 - [x] How should nested groups work? (graph subgroups or graph meters? Optional? Default?)

Initially, focus should be on comparing groups with each other.

**Comment by LeoTindall** [comment]
In my opinion, the best experience is obtained if groups can be selected as items (like meters are now) and graphed with all meters and sub-groups in aggregate. Sub-meters can be selected individual or in groups.

It would be good to add a way to select all sub-meters; perhaps a button saying "break down into sub-units" or something. This would be a tree operation, not a flat-map, so:

```
A -+
     |
     + - B
     + - C
     + - D
           + - E
           + - F
      + - G
H - +
       |
       + - I
```

Selecting A would graph the aggregate of B C and D (which would in turn be the aggregate of E and F), but not H (or, by extension, I).

Then, selecting "break into sub units" would separately graph B, C, and D (which is the aggregate of E and F). Selecting "break into sub units" _again_ would separately graph B, C, E, and F. The user could manually select A and D again if they wished.

Does this seem like a good idea?

**Comment by sandeepacharya464** [comment]
I like the idea of displaying one layer of sub-units at a time. This makes it lot easier for the user to figure out the immediate parent and children of various groups and sub-groups.

**Comment by LeoTindall** [comment]
@johndiiorio mentioned that we may want groups to be opaque to the user. In that case, I propose the following changes:

* Show groups in the `meterList` along with meters
* Clicking a group in the `meterList` has the same effect as clicking a meter
* Groups can only be managed from the groups management UI

This results in hiding the DAG from the user, which vastly simplifies the UI.

**Comment by huss** [comment]
I propose we discuss this at our team meeting. It is also likely to be an
item where doing agile development by reaching out to end users for ideas
and feedback during development would be useful.

**Comment by LeoTindall** [comment]
@johndiiorio says: Groups have a name field, perhaps add a group description field in a tooltip and an annotation of how many meters.

Also, add an annotation link that allows users to go to group management.

After this, ship to users and see if they need a drill-down button.

**Comment by johndiiorio** [comment]
Implemented in #148
